### PR TITLE
add controllers for import and stat

### PR DESCRIPTION
**Pull request by BelehovEgor**

### CODE REVIEW

**Review by MariyaSin — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

**Review by VLiamin — APPROVED** [review]

**Review by borovHokage — APPROVED** [review]

File: src/TimeService.Models.Dto/Filters/FindStatFilter.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using Microsoft.AspNetCore.Mvc;
+
+namespace LT.DigitalOffice.TimeService.Models.Dto.Filters
+{
+  public record FindStatFilter
+  {
+    [FromQuery(Name = "departmentId")]
```
Comment (djino97) [inline src/TimeService.Models.Dto/Filters/FindStatFilter.cs:8]: departmentid

File: src/TimeService.Models.Dto/Filters/ImportStatFilter.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using Microsoft.AspNetCore.Mvc;
+
+namespace LT.DigitalOffice.TimeService.Models.Dto.Filters
+{
+  public record ImportStatFilter
+  {
+    [FromQuery(Name = "departmentId")]
```
Comment (djino97) [inline src/TimeService.Models.Dto/Filters/ImportStatFilter.cs:8]: departmentid

**Review by djino97 — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

File: src/TimeService.Business/Commands/LeaveTime/FindLeaveTimesCommand.cs [hunk 3]
```diff
@@ -1,60 +1,113 @@
...
+    }
+
+    public FindResultResponse<LeaveTimeResponse> Execute(FindLeaveTimesFilter filter)
+    {
+      if (filter == null)
+      {
+        throw new ArgumentNullException(nameof(filter));
+      }
+
+      var isActhor = filter.UserId.HasValue && filter.UserId == _httpContextAccessor.HttpContext.GetUserId();
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/TimeService.Business/Commands/LeaveTime/FindLeaveTimesCommand.cs]: fixed

File: src/TimeService.Business/Commands/Stat/FindStatCommand.cs [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+        _logger.LogWarning(loggerMessage + "Reasons: {Errors}", string.Join("\n", response.Message.Errors));
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, loggerMessage);
+      }
+
+      errors.Add(message);
+
+      return new();
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/TimeService.Business/Commands/Stat/FindStatCommand.cs]: fixed

File: src/TimeService.Business/Commands/Stat/FindStatCommand.cs [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+
+      if (filter.DepartmentId.HasValue)
+      {
+        projectUsers = GetProjectsUsers(userIds, errors);
+      }
+      else
+      {
+        projectUsers = projects?.SelectMany(p => p.Users).ToList();
+
+        userIds = projectUsers?.Select(pu => pu.UserId).OrderBy(id => id).Distinct().ToList();
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/TimeService.Business/Commands/Stat/FindStatCommand.cs]: removed

File: src/TimeService.Business/Helpers/Workdays/WorkTimeLimitCreater.cs [hunk 6]
```diff
@@ -1,104 +1,109 @@
...
-            }
+      DbWorkTimeMonthLimit limit = new()
+      {
+        Id = Guid.NewGuid(),
+        ModifiedAtUtc = DateTime.UtcNow,
+        Holidays = holidays,
+        ModifiedBy = null,
+        Month = time.Month,
+        Year = time.Year,
+        NormHours = holidays.ToCharArray().Count(h => h == '0') * 8
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/TimeService.Business/Helpers/Workdays/WorkTimeLimitCreater.cs]: created

File: src/TimeService.Data/LeaveTimeRepository.cs [hunk 7]
```diff
@@ -1,93 +1,108 @@
...
+
+      if (filter.EndTime.HasValue)
+      {
+        dbLeaveTimes = dbLeaveTimes.Where(x => x.EndTime <= filter.EndTime || x.StartTime < filter.EndTime);
+      }
+
+      if (!(filter.IncludeDeactivated.HasValue && filter.IncludeDeactivated.Value))
+      {
+        dbLeaveTimes = dbLeaveTimes.Where(x => x.IsActive);
+      }
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/TimeService.Data/LeaveTimeRepository.cs]: created new method

File: src/TimeService.Models.Dto/Filters/ImportStatFilter.cs (hunk 2 above)
Comment (BelehovEgor) [reply src/TimeService.Models.Dto/Filters/ImportStatFilter.cs:8]: fixed

File: src/TimeService.Models.Dto/Filters/FindStatFilter.cs (hunk 1 above)
Comment (BelehovEgor) [reply src/TimeService.Models.Dto/Filters/FindStatFilter.cs]: fixed

**Pull request merged**